camel-debezium-mysql - Integration test fails due to an access denied

The integration test of camel-debezium-mysql fails due to an access defined as show in the following log message:
```
2025-01-21T13:03:58.3948863Z org.springframework.jdbc.CannotGetJdbcConnectionException: Failed to obtain JDBC Connection
2025-01-21T13:03:58.3950092Z 	at org.springframework.jdbc.datasource.DataSourceUtils.getConnection(DataSourceUtils.java:84) ~[?:?]
2025-01-21T13:03:58.3951237Z 	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:653) ~[?:?]
2025-01-21T13:03:58.3952238Z 	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:695) ~[?:?]
2025-01-21T13:03:58.3953244Z 	at org.apache.camel.component.sql.SqlProducer.processInternal(SqlProducer.java:150) ~[?:?]
2025-01-21T13:03:58.3954241Z 	at org.apache.camel.component.sql.SqlProducer.process(SqlProducer.java:137) ~[?:?]
2025-01-21T13:03:58.3955806Z 	at org.apache.camel.support.AsyncProcessorConverterHelper$ProcessorToAsyncProcessorBridge.process(AsyncProcessorConverterHelper.java:65) ~[?:?]
2025-01-21T13:03:58.3957223Z 	at org.apache.camel.processor.SendProcessor.process(SendProcessor.java:208) ~[?:?]
2025-01-21T13:03:58.3958769Z 	at org.apache.camel.processor.errorhandler.RedeliveryErrorHandler$SimpleTask.handleFirst(RedeliveryErrorHandler.java:440) ~[?:?]
2025-01-21T13:03:58.3960305Z 	at org.apache.camel.processor.errorhandler.RedeliveryErrorHandler$SimpleTask.run(RedeliveryErrorHandler.java:416) ~[?:?]
2025-01-21T13:03:58.3961656Z 	at org.apache.camel.impl.engine.DefaultReactiveExecutor$Worker.doRun(DefaultReactiveExecutor.java:199) ~[?:?]
2025-01-21T13:03:58.3963537Z 	at org.apache.camel.impl.engine.DefaultReactiveExecutor$Worker.executeReactiveWork(DefaultReactiveExecutor.java:189) ~[?:?]
2025-01-21T13:03:58.3965074Z 	at org.apache.camel.impl.engine.DefaultReactiveExecutor$Worker.tryExecuteReactiveWork(DefaultReactiveExecutor.java:166) ~[?:?]
2025-01-21T13:03:58.3966543Z 	at org.apache.camel.impl.engine.DefaultReactiveExecutor$Worker.schedule(DefaultReactiveExecutor.java:148) ~[?:?]
2025-01-21T13:03:58.3968262Z 	at org.apache.camel.impl.engine.DefaultReactiveExecutor.scheduleMain(DefaultReactiveExecutor.java:59) ~[?:?]
2025-01-21T13:03:58.3969139Z 	at org.apache.camel.processor.Pipeline.process(Pipeline.java:163) ~[?:?]
2025-01-21T13:03:58.3969810Z 	at org.apache.camel.impl.engine.CamelInternalProcessor.processNonTransacted(CamelInternalProcessor.java:347) ~[?:?]
2025-01-21T13:03:58.3970573Z 	at org.apache.camel.impl.engine.CamelInternalProcessor.process(CamelInternalProcessor.java:323) ~[?:?]
2025-01-21T13:03:58.3971225Z 	at org.apache.camel.component.direct.DirectProducer.process(DirectProducer.java:102) ~[?:?]
2025-01-21T13:03:58.3971992Z 	at org.apache.camel.impl.engine.SharedCamelInternalProcessor.processNonTransacted(SharedCamelInternalProcessor.java:156) ~[?:?]
2025-01-21T13:03:58.3972844Z 	at org.apache.camel.impl.engine.SharedCamelInternalProcessor.process(SharedCamelInternalProcessor.java:133) ~[?:?]
2025-01-21T13:03:58.3973630Z 	at org.apache.camel.impl.engine.SharedCamelInternalProcessor$1.process(SharedCamelInternalProcessor.java:89) ~[?:?]
2025-01-21T13:03:58.3974449Z 	at org.apache.camel.impl.engine.DefaultAsyncProcessorAwaitManager.process(DefaultAsyncProcessorAwaitManager.java:82) ~[?:?]
2025-01-21T13:03:58.3975280Z 	at org.apache.camel.impl.engine.SharedCamelInternalProcessor.process(SharedCamelInternalProcessor.java:86) ~[?:?]
2025-01-21T13:03:58.3975998Z 	at org.apache.camel.support.cache.DefaultProducerCache.send(DefaultProducerCache.java:178) ~[?:?]
2025-01-21T13:03:58.3976662Z 	at org.apache.camel.impl.engine.DefaultProducerTemplate.send(DefaultProducerTemplate.java:176) ~[?:?]
2025-01-21T13:03:58.3977329Z 	at org.apache.camel.impl.engine.DefaultProducerTemplate.send(DefaultProducerTemplate.java:172) ~[?:?]
2025-01-21T13:03:58.3978292Z 	at org.apache.camel.impl.engine.DefaultProducerTemplate.send(DefaultProducerTemplate.java:153) ~[?:?]
2025-01-21T13:03:58.3979183Z 	at org.apache.karaf.camel.itests.CamelSingleFeatureRoute.triggerProducerRoute(CamelSingleFeatureRoute.java:59) ~[?:?]
2025-01-21T13:03:58.3980238Z 	at org.apache.karaf.camel.itests.AbstractCamelSingleFeatureResultMockBasedRouteITest.setupMock(AbstractCamelSingleFeatureResultMockBasedRouteITest.java:29) ~[?:?]
2025-01-21T13:03:58.3981130Z 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
2025-01-21T13:03:58.3981719Z 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
2025-01-21T13:03:58.3982408Z 	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
2025-01-21T13:03:58.3982949Z 	at java.lang.reflect.Method.invoke(Method.java:569) ~[?:?]
2025-01-21T13:03:58.3983431Z 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50) ~[?:?]
2025-01-21T13:03:58.3984043Z 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) ~[?:?]
2025-01-21T13:03:58.3984719Z 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47) ~[?:?]
2025-01-21T13:03:58.3985318Z 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24) ~[?:?]
2025-01-21T13:03:58.3985944Z 	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) ~[?:?]
2025-01-21T13:03:58.3986510Z 	at org.apache.karaf.camel.itests.DumpFileOnError$1.evaluate(DumpFileOnError.java:45) ~[?:?]
2025-01-21T13:03:58.3987013Z 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55) ~[?:?]
2025-01-21T13:03:58.3987522Z 	at org.apache.karaf.itests.KarafTestSupport$Retry$1.evaluate(KarafTestSupport.java:160) ~[?:?]
2025-01-21T13:03:58.3988326Z 	at org.junit.rules.RunRules.evaluate(RunRules.java:20) ~[?:?]
2025-01-21T13:03:58.3988923Z 	at org.ops4j.pax.exam.invoker.junit.internal.ContainerTestRunner.runLeafWithRetry(ContainerTestRunner.java:97) ~[?:?]
2025-01-21T13:03:58.3989745Z 	at org.ops4j.pax.exam.invoker.junit.internal.ContainerTestRunner.runChildWithRetry(ContainerTestRunner.java:84) ~[?:?]
2025-01-21T13:03:58.3990533Z 	at org.ops4j.pax.exam.invoker.junit.internal.ContainerTestRunner.runChild(ContainerTestRunner.java:75) ~[?:?]
2025-01-21T13:03:58.3991402Z 	at org.ops4j.pax.exam.invoker.junit.internal.ContainerTestRunner.runChild(ContainerTestRunner.java:43) ~[?:?]
2025-01-21T13:03:58.3992026Z 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290) ~[?:?]
2025-01-21T13:03:58.3992471Z 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71) ~[?:?]
2025-01-21T13:03:58.3992940Z 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288) ~[?:?]
2025-01-21T13:03:58.3993403Z 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58) ~[?:?]
2025-01-21T13:03:58.3993860Z 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268) ~[?:?]
2025-01-21T13:03:58.3994304Z 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363) ~[?:?]
2025-01-21T13:03:58.3994700Z 	at org.junit.runner.JUnitCore.run(JUnitCore.java:137) ~[?:?]
2025-01-21T13:03:58.3995076Z 	at org.junit.runner.JUnitCore.run(JUnitCore.java:115) ~[?:?]
2025-01-21T13:03:58.3995646Z 	at org.ops4j.pax.exam.invoker.junit.internal.JUnitProbeInvoker.invokeViaJUnit(JUnitProbeInvoker.java:124) ~[?:?]
2025-01-21T13:03:58.3996413Z 	at org.ops4j.pax.exam.invoker.junit.internal.JUnitProbeInvoker.findAndInvoke(JUnitProbeInvoker.java:97) ~[?:?]
2025-01-21T13:03:58.3997126Z 	at org.ops4j.pax.exam.invoker.junit.internal.JUnitProbeInvoker.call(JUnitProbeInvoker.java:73) ~[?:?]
2025-01-21T13:03:58.3997747Z 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
2025-01-21T13:03:58.3998527Z 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
2025-01-21T13:03:58.3999221Z 	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
2025-01-21T13:03:58.3999764Z 	at java.lang.reflect.Method.invoke(Method.java:569) ~[?:?]
2025-01-21T13:03:58.4000478Z 	at org.ops4j.pax.exam.rbc.internal.RemoteBundleContextImpl.remoteCall(RemoteBundleContextImpl.java:85) ~[!/:?]
2025-01-21T13:03:58.4001127Z 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
2025-01-21T13:03:58.4001713Z 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
2025-01-21T13:03:58.4002396Z 	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
2025-01-21T13:03:58.4002929Z 	at java.lang.reflect.Method.invoke(Method.java:569) ~[?:?]
2025-01-21T13:03:58.4003355Z 	at sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:360) ~[?:?]
2025-01-21T13:03:58.4003798Z 	at sun.rmi.transport.Transport$1.run(Transport.java:200) ~[?:?]
2025-01-21T13:03:58.4004178Z 	at sun.rmi.transport.Transport$1.run(Transport.java:197) ~[?:?]
2025-01-21T13:03:58.4004626Z 	at java.security.AccessController.doPrivileged(AccessController.java:712) [?:?]
2025-01-21T13:03:58.4005169Z 	at sun.rmi.transport.Transport.serviceCall(Transport.java:196) ~[?:?]
2025-01-21T13:03:58.4005662Z 	at sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:592) ~[?:?]
2025-01-21T13:03:58.4006207Z 	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:834) ~[?:?]
2025-01-21T13:03:58.4006883Z 	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(TCPTransport.java:711) ~[?:?]
2025-01-21T13:03:58.4007454Z 	at java.security.AccessController.doPrivileged(AccessController.java:399) [?:?]
2025-01-21T13:03:58.4008143Z 	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:710) [?:?]
2025-01-21T13:03:58.4008716Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
2025-01-21T13:03:58.4009286Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
2025-01-21T13:03:58.4009711Z 	at java.lang.Thread.run(Thread.java:840) [?:?]
2025-01-21T13:03:58.4010170Z Caused by: java.sql.SQLException: Access denied for user 'root'@'172.17.0.1' (using password: YES)
2025-01-21T13:03:58.4010740Z 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:121) ~[?:?]
2025-01-21T13:03:58.4011474Z 	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:113) ~[?:?]
2025-01-21T13:03:58.4012094Z 	at com.mysql.cj.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:861) ~[?:?]
2025-01-21T13:03:58.4012583Z 	at com.mysql.cj.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:449) ~[?:?]
2025-01-21T13:03:58.4013061Z 	at com.mysql.cj.jdbc.ConnectionImpl.getInstance(ConnectionImpl.java:234) ~[?:?]
2025-01-21T13:03:58.4013582Z 	at com.mysql.cj.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:180) ~[?:?]
2025-01-21T13:03:58.4014129Z 	at com.mysql.cj.jdbc.MysqlDataSource.getConnection(MysqlDataSource.java:408) ~[?:?]
2025-01-21T13:03:58.4014657Z 	at com.mysql.cj.jdbc.MysqlDataSource.getConnection(MysqlDataSource.java:120) ~[?:?]
2025-01-21T13:03:58.4015174Z 	at com.mysql.cj.jdbc.MysqlDataSource.getConnection(MysqlDataSource.java:105) ~[?:?]
2025-01-21T13:03:58.4015791Z 	at org.springframework.jdbc.datasource.DataSourceUtils.fetchConnection(DataSourceUtils.java:160) ~[?:?]
2025-01-21T13:03:58.4016506Z 	at org.springframework.jdbc.datasource.DataSourceUtils.doGetConnection(DataSourceUtils.java:118) ~[?:?]
2025-01-21T13:03:58.4017215Z 	at org.springframework.jdbc.datasource.DataSourceUtils.getConnection(DataSourceUtils.java:81) ~[?:?]
```

--- a/tests/features/camel-debezium-mysql/pom.xml
+++ b/tests/features/camel-debezium-mysql/pom.xml
@@ -51,4 +51,40 @@
             <version>${camel-version}</version>
         </dependency>
     </dependencies>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>default-test</id>
+                        <goals>
+                            <goal>test</goal>
+                        </goals>
+                        <phase>integration-test</phase>
+                        <configuration>
+                            <excludes>
+                                <exclude>none</exclude>
+                            </excludes>
+                            <includes>
+                                <include>**/*Test.java</include>
+                            </includes>
+                            <systemPropertyVariables>
+                                <debezium.version>${debezium-version}</debezium.version>
+                                <camel.karaf.itest.dump.logs>${dump.logs.on.failure}</camel.karaf.itest.dump.logs>
+                                <camel.karaf.itest.keep.docker.images>${keep.docker.images.on.exit}</camel.karaf.itest.keep.docker.images>
+                                <camel.karaf.version>${project.version}</camel.karaf.version>
+                                <project.version>${project.version}</project.version>
+                                <project.target>${project.build.directory}</project.target>
+                                <users.file.location>${users.file.location}</users.file.location>
+                                <org.ops4j.pax.logging.DefaultServiceLog.level>WARN</org.ops4j.pax.logging.DefaultServiceLog.level>
+                            </systemPropertyVariables>
+                            <forkedProcessExitTimeoutInSeconds>10</forkedProcessExitTimeoutInSeconds>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
 </project>

--- a/tests/features/camel-debezium-mysql/src/test/java/org/apache/karaf/camel/itest/CamelDebeziumMysqlITest.java
+++ b/tests/features/camel-debezium-mysql/src/test/java/org/apache/karaf/camel/itest/CamelDebeziumMysqlITest.java
@@ -45,7 +45,6 @@ public class CamelDebeziumMysqlITest extends AbstractCamelSingleFeatureResultMoc
 
     public static final class ExternalResourceProviders {
 
-        private static final String DEBEZIUM_VERSION = "2.7";
         private static final String MYSQL_IMAGE = "quay.io/debezium/example-mysql";
         private static final String SOURCE_DB_NAME = "inventory";
         private static final String SOURCE_DB_TABLE = String.format("%s.products", SOURCE_DB_NAME);
@@ -54,7 +53,7 @@ public class CamelDebeziumMysqlITest extends AbstractCamelSingleFeatureResultMoc
 
         public static GenericContainerResource<MYSQLContainer> createMySQLContainer() {
 
-            MYSQLContainer container = new MYSQLContainer(DockerImageName.parse(MYSQL_IMAGE).withTag(DEBEZIUM_VERSION)
+            MYSQLContainer container = new MYSQLContainer(DockerImageName.parse(MYSQL_IMAGE).withTag(System.getProperty("debezium.version"))
                     .asCompatibleSubstituteFor("mysql"))
                     .withUsername(SOURCE_DB_USERNAME)
                     .withPassword(SOURCE_DB_PASSWORD);

